auth/realm/base/src/main/java/org/wildfly/security/auth/realm/AggregateSecurityRealm.java

```diff
@@ -67,8 +77,11 @@ public RealmIdentity getRealmIdentity(final Evidence evidence) throws RealmUnava
         try {
             for (int i = 0; i < authorizationIdentities.length; i++) {
                 SecurityRealm authorizationRealm = authorizationRealms[i];
-
-                authorizationIdentities[i] = authorizationRealm == authenticationRealm ? authenticationIdentity : authorizationRealm.getRealmIdentity(evidence);
+                if(principalTransformer == null) {
```

Comment:
Just a very minor code style comment, there should be a white space after each `if` before the opening bracket.

Reply:
Fixed